apps/site/src/pages/settings/billing-settings-panel/usage-limit-section.tsx

```diff
@@ -0,0 +1,195 @@
+  const { user } = useUser();
+
+  const [usageLimitValue, setUsageLimitValue] = useState("");
+
+  const [actionState, setActionState] = useState<ActionState>({
+    state: "loading",
+  });
+  const [usageLimit, setUsageLimit] = useState<UsageLimitState | null>(null);
+
+  useEffect(() => {
+    if (
+      user &&
+      user !== "loading" &&
+      user.usageLimitCents &&
+      Number.isInteger(user.usageLimitCents)
+    ) {
+      setUsageLimitValue(
+        priceToHumanReadable({
+          amountInCents: user.usageLimitCents,
+          currency: "",
+        }),
+      );
+    } else {
+      setUsageLimitValue("");
+    }
+
+    setActionState({ state: "idle" });
+  }, [user, setActionState]);
+
+  const handleLimitChange = ({
+    target: { value },
+  }: ChangeEvent<HTMLInputElement>) => {
+    if (usageLimitUsdRegex.test(value)) {
+      setUsageLimitValue(value);
+
+      const newUsageLimit: UsageLimitState = value
+        ? { limit: "capped", usageLimitCents: convertToCents(value) }
+        : { limit: "uncapped" };
+
+      setUsageLimit(newUsageLimit);
+      setActionState({ state: "idle" });
+    }
+  };
+
+  const debouncedUsageLimit = useMemo(
+    () =>
+      debounce(async (params: { usageLimitState: UsageLimitState }) => {
+        const { usageLimitState } = params;
+
+        setActionState({ state: "loading" });
+
+        const { error } = await apiClient.setUsageLimit(usageLimitState);
+
+        if (error) {
+          setActionState({
+            state: "error",
+            message: "An error occurred, please try again with a valid value.",
+          });
+        } else {
+          setActionState({ state: "success" });
+        }
+
+        setUsageLimit(null);
+      }, 500),
+    [],
+  );
+
+  useEffect(() => {
+    if (actionState.state === "loading" || usageLimit === null) {
+      return;
+    }
+
+    void debouncedUsageLimit({
+      usageLimitState: usageLimit,
+    });
+  }, [debouncedUsageLimit, usageLimit, actionState]);
+
+  return (
+    <>
+      <Typography
+        variant="bpHeading2"
+        sx={{ fontSize: 28, fontWeight: 400, marginBottom: 1 }}
+      >
+        Usage Limits
+      </Typography>
+      <Typography
+        component="p"
+        variant="bpSmallCopy"
+        sx={{
+          color: ({ palette }) => palette.gray[90],
+          opacity: 0.66,
+          marginBottom: 2,
+        }}
+      >
+        Optionally cap usage-based charges. Leave blank to uncap, or enter $0 to
+        prevent all overage charges.
+      </Typography>
+      <Typography
+        htmlFor="usage-limit-input"
+        component="label"
+        variant="bpSmallCopy"
+        sx={{
+          color: ({ palette }) => palette.gray[90],
+          opacity: 0.66,
+          // marginBottom: 1,
+        }}
+      >
+        Spending cap
+      </Typography>
+      <TextField
+        id="usage-limit-input"
+        sx={{
+          width: 375,
```

Comment:
We might want this to shrink in smaller screens
```suggestion
          maxWidth: 375,
```

Reply:
That makes sense 859f597f1334ba28d1870350f10470688afc9f8d